framework/Optimizers/GradientDescent.py

```diff
@@ -0,0 +1,807 @@
+                            Note that gradient descent algorithms are particularly prone to being trapped
+                            in local minima; for this reason, depending on the model, multiple trajectories
+                            may be needed to obtain the global solution.
+                            """
+    # gradient estimation options
+    grad = InputData.parameterInputFactory('gradient', strictMode=True,
+        printPriority=106,
+        descr=r"""a required node containing the information about which gradient approximation algorithm to
+              use, and its settings if applicable. Exactly one of the gradient approximation algorithms
+              below may be selected for this Optimizer.""")
+    specs.addSub(grad)
+    ## get specs for each gradient subclass, and add them to this class's options
+    for option in gradKnownTypes():
+      subSpecs = gradReturnClass(option, cls).getInputSpecification()
+      grad.addSub(subSpecs)
+
+    # step sizing options
+    step = InputData.parameterInputFactory('stepSize', strictMode=True,
+        printPriority=107,
+        descr=r"""a required node containing the information about which iterative stepping algorithm to
+              use, and its settings if applicable. Exactly one of the stepping algorithms
+              below may be selected for this Optimizer.""")
+    specs.addSub(step)
+    ## common options to all stepManipulator descenders
+    ## TODO
+    ## get specs for each stepManipulator subclass, and add them to this class's options
+    for option in stepKnownTypes():
+      subSpecs = stepReturnClass(option, cls).getInputSpecification()
+      step.addSub(subSpecs)
+
+    # acceptance conditions
+    accept = InputData.parameterInputFactory('acceptance', strictMode=True,
+        printPriority=108,
+        descr=r"""a required node containing the information about the acceptability criterion for iterative
+              optimization steps, i.e. when a potential new optimal point should be rejected and when
+              it can be accepted. Exactly one of the acceptance criteria
+              below may be selected for this Optimizer.""")
+    specs.addSub(accept)
+    ## common options to all acceptanceCondition descenders
+    ## TODO
+    ## get specs for each acceptanceCondition subclass, and add them to this class's options
+    for option in acceptKnownTypes():
+      subSpecs = acceptReturnClass(option, cls).getInputSpecification()
+      accept.addSub(subSpecs)
+
+    # convergence
+    conv = InputData.parameterInputFactory('convergence', strictMode=True,
+        printPriority=109,
+        descr=r"""a node containing the desired convergence criteria for the optimization algorithm.
+              Note that convergence is met when any one of the convergence criteria is met. If no convergence
+              criteria are given, then the \xmlNode{limit} is used.""")
```

Comment:
Shouldn't the defaults be used first before the limit?

Reply:
fixed! thanks